src/pages/admin/categories/CategoryList/index.js

```diff
@@ -79,7 +104,20 @@ const CategoryList = () => {
               </Link>
             </td>
             <td>
-              <Button buttonLabel="Delete" buttonSize="sm" outline={true} />
+              <Button
+                buttonLabel="Delete"
+                buttonSize="sm"
+                outline={true}
+                onClick={() => {
+                  if (category.subcategories_count <= 0) {
+                    setCanDelete(true);
+                  } else {
+                    setCanDelete(false);
+                  }
+                  setItemToDelete(category);
+                  setShowConfirmationModal(true);
+                }}
```

Comment:
move to a function. it looks similar to my previous comment. maybe you can use the same function?